src/Diagnostics.ModelsAndUtils/ScriptUtilities/Utilities.cs

```diff
@@ -116,18 +116,33 @@ public static string HostNamesFilterQuery(IEnumerable<string> hostnames, string
 
             if (nonWildCardHostNames.Any())
             {
+                nonWildCardHostNames = AddPortNumberToHostNames(nonWildCardHostNames);
                 hostNameQuery = $"{hostNameColumn} in~ ({string.Join(",", nonWildCardHostNames.Select(h => $@"""{h}"""))})";
             }
 
             if (wildCardHostNames.Any())
             {
-                string wildCardQuery = string.Join("or", wildCardHostNames.Select(w => $@" {hostNameColumn} endswith ""{w.Replace("*.", ".")}"""));
+                wildCardHostNames = AddPortNumberToHostNames(wildCardHostNames);
+                string wildCardQuery = string.Join(" or", wildCardHostNames.Select(w => $@" {hostNameColumn} endswith ""{w.Replace("*.", ".")}"""));
                 hostNameQuery = $"{hostNameQuery} or {wildCardQuery}";
             }
 
             return hostNameQuery;
         }
 
+        private static IEnumerable<string> AddPortNumberToHostNames(IEnumerable<string> hostnames)
+        {
+            var hostNamesWithPort = new List<string>();
+            foreach (var host in hostnames)
+            {
+                hostNamesWithPort.Add(host);
```

Comment:
What's the possibility of a "dirty" input, something with a port already on it? Maybe to clean this you can start with a base hostname of `host.Split(':').First()`

Reply:
most of the detectors are just passing cxt.Resource.Hostnames which should just be the hostnames configured on the app so the possibility of a dirty input is quite less. Worst case we will add the original input and another dirty input with the port twice in it but the overall logic of the KUSTO query should still be fine